proto-compiler/src/main.rs

```diff
@@ -1,36 +1,129 @@
+use std::env::var;
 use std::fs::remove_dir_all;
 use std::fs::{copy, create_dir_all};
+use std::path::{Path, PathBuf};
+
+use git2::Repository;
+use tempdir::TempDir;
 use walkdir::WalkDir;
 
-pub(crate) fn main() {
-    let ibc_proto_path = "../proto/src/prost";
+fn main() {
+    let root = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+    let target_dir = root.join("../proto/src/prost");
+    let out_dir = var("OUT_DIR")
+        .map(PathBuf::from)
+        .or_else(|_| TempDir::new("ibc_proto_out").map(|d| d.into_path()))
+        .unwrap();
+
+    let sdk_dir = clone_cosmos_sdk();
+    compile_protos(&sdk_dir, &out_dir);
+    copy_generated_files(&out_dir, &target_dir);
+}
+
+fn clone_cosmos_sdk() -> PathBuf {
+    let sdk_dir = var("SDK_DIR").unwrap_or_else(|_| "target/cosmos-sdk".to_string());
+
+    if Path::new(&sdk_dir).exists() {
+        println!("[info ] Found Cosmos SDK source at '{}'", sdk_dir);
+    } else {
+        println!("[info ] Cloning cosmos/cosmos-sdk repository...");
+
+        let url = "https://github.com/cosmos/cosmos-sdk";
+        Repository::clone(url, &sdk_dir).unwrap();
```

Comment:
I'll have a more robust solution to this in the tendermint-proto crate. Later, for a next update.